src/Common/EventRateMeter.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <base/defines.h>
+#include <base/types.h>
+#include <algorithm>
+#include <cmath>
+#include <cassert>
+
+namespace DB
+{
+
+/// Event count measurement with exponential smoothing intended for computing time derivatives
+class EventRateMeter {
```

Comment:
Maybe reuse Common/ExponentialSmoothedCounter.h?

Reply:
Nice, I did not know we have it and it is even better generalised! Let me think how to use it inside EventRateMeter. My implementation is almost the same, but has small advantage at start - it will converge faster due to expression in denominator inside `rate()`. But anyway I think we can combine all that.

Reply:
done